lesson 24_hw_sign-in-up/users/views.py

```diff
@@ -0,0 +1,72 @@
+from django.shortcuts import render, redirect
+from django.views.generic.base import View
+from django.contrib import messages
+from .models import Users
+from users.additional_funcs.get_data import (
+    get_all_nicknames,
+    get_all_emails,
+    hide_email,
+)
+from users.additional_funcs.pwd_actions import hash_password, check_password
+
+
+class MainView(View):
+    def get(self, request):
+        users = Users.objects.values_list("email", flat=True)
+        users_count = len(users)
+        hidden_email = hide_email(users)
+        return render(
+            request, "main.html", {"count": users_count, "hidden_emails": hidden_email}
+        )
+
+
+class RegistrationView(View):
+    def get(self, request):
+        return render(request, "signup.html")
+
+    def post(self, request):
+        email = request.POST["email"].lower()
+        password = request.POST["password"]
+        password_2 = request.POST["password_2"]
+        nickname = request.POST["nickname"]
+
+        used_email = get_all_emails()
+        used_nickname = get_all_nicknames()
+
+        if email and nickname and email not in used_email and nickname not in used_nickname and password == password_2:

```

Comment:
It is too hard to read this line.. try to simplify it

Reply:
Better to try `Users.objects.get` some user with passed email and catch DoesNotExist exception